src/core/meshcop/tcat_agent.cpp

```diff
@@ -0,0 +1,515 @@
+        mCommissionerHasDomainName      = true;
+    }
+
+    len = sizeof(mCommissionerNetworkName) - 1;
+    if (aTlsContext.GetThreadAttributeFromPeerCertificate(
+            kCertificateNetworkName, reinterpret_cast<uint8_t *>(&mCommissionerNetworkName), &len) == kErrorNone)
+    {
+        mCommissionerNetworkName.m8[len] = '\0';
+        mCommissionerHasNetworkName      = true;
+    }
+
+    len = sizeof(mCommissionerExtendedPanId);
+    if (aTlsContext.GetThreadAttributeFromPeerCertificate(
+            kCertificateExtendedPanId, reinterpret_cast<uint8_t *>(&mCommissionerExtendedPanId), &len) == kErrorNone)
+    {
+        if (len == sizeof(mCommissionerExtendedPanId))
+        {
+            mCommissionerHasExtendedPanId = true;
+        }
+    }
+
+    mCurrentApplicationLayer = kMessageTypeNone;
+    mCurrentServiceName[0]   = 0;
+    mState                   = kStateConnected;
+    mAlreadyCommissioned     = Get<ActiveDatasetManager>().IsCommissioned();
+    LogInfo("TCAT agent connected");
+
+exit:
+    return error;
+}
+
+void TcatAgent::Disconnected(void)
+{
+    mCurrentApplicationLayer = kMessageTypeNone;
+    mAlreadyCommissioned     = false;
+
+    if (mState != kStateDisabled)
+    {
+        mState = kStateEnabled;
+    }
+
+    LogInfo("TCAT agent disconnected");
+}
+
+bool TcatAgent::IsCommandClassAuthorized(CommandClassFlags aCommissionerCommandClassFlags,
+                                         CommandClassFlags aDeviceCommandClassFlags,
+                                         Dataset          *aDataset) const
+{
+    bool authorized                     = false;
+    bool additionalDeviceRequirementMet = false;
+    bool domainNamesMatch               = false;
+    bool networkNamesMatch              = false;
+    bool extendedPanIdsMatch            = false;
+
+    VerifyOrExit(IsConnected());
+    VerifyOrExit(aCommissionerCommandClassFlags & kAccessFlag);
+
+    if (aDeviceCommandClassFlags & kAccessFlag)
+    {
+        additionalDeviceRequirementMet = true;
+    }
+
+    if (aDeviceCommandClassFlags & kPskdFlag)
+    {
+        additionalDeviceRequirementMet = true;
+    }
+
+    if (aDeviceCommandClassFlags & kPskcFlag)
+    {
+        additionalDeviceRequirementMet = true;
+    }
+
+    if (mCommissionerHasNetworkName || mCommissionerHasExtendedPanId)
+    {
+        Dataset::Info datasetInfo;
+        Error         datasetError = kErrorNone;
+
+        if (aDataset == nullptr)
+        {
+            datasetError = Get<ActiveDatasetManager>().Read(datasetInfo);
+        }
+        else
+        {
+            aDataset->ConvertTo(datasetInfo);
+        }
+
+        if (datasetError == kErrorNone)
+        {
+            if (datasetInfo.IsNetworkNamePresent() && mCommissionerHasNetworkName)
+            {
+                if (datasetInfo.GetNetworkName() == mCommissionerNetworkName)
+                {
+                    networkNamesMatch = true;
+                }
+            }
```

Comment:
Combine nested `if`s into one `if`?
```suggestion
            if (datasetInfo.IsNetworkNamePresent() && mCommissionerHasNetworkName & 
 (datasetInfo.GetNetworkName() == mCommissionerNetworkName)
                {
                    networkNamesMatch = true;
                }
 ```

Reply:
fixed.